modules/asset-manager-storage-aws/src/main/java/org/opencastproject/assetmanager/aws/s3/AwsS3AssetStore.java

```diff
@@ -143,10 +143,10 @@ public void activate(final ComponentContext cc) throws IllegalStateException, IO
       regionName = getAWSConfigKey(cc, AWS_S3_REGION_CONFIG);
       logger.info("AWS region is {}", regionName);
 
-      endpoint = getAWSConfigKey(cc, AWS_S3_ENDPOINT_CONFIG);
+      endpoint = OsgiUtil.getComponentContextProperty(cc, AWS_S3_ENDPOINT_CONFIG, "s3." + regionStr + ".amazonaws.com");
       logger.info("AWS endpoint is {}", endpoint);
 
-      pathStyle = Boolean.valueOf(getAWSConfigKey(cc, AWS_S3_PATH_STYLE_CONFIG));
+      pathStyle = Boolean.valueOf(OsgiUtil.getComponentContextProperty(cc, AWS_S3_PATH_STYLE_CONFIG, "false"));
```

Comment:
We use [BooleanUtils.toBoolean](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/BooleanUtils.html#toBoolean-java.lang.String-) at a lot of other places. It is much more robust when it comes to casing, …

```suggestion
      pathStyle = BooleanUtils.toBoolean(OsgiUtil.getComponentContextProperty(cc, AWS_S3_PATH_STYLE_CONFIG, "false"));
```